File: backend/app/schemas/notification.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List, Dict, Any
from datetime import datetime
from uuid import UUID
from app.models.messaging import (
    NotificationType, NotificationPriority, NotificationCategory,
    NotificationAction, EntityType
)

class NotificationBase(BaseModel):
    title: str
    body: str
    link: Optional[str] = None
    notification_type: NotificationType
    priority: NotificationPriority
    category: NotificationCategory
    entity_type: Optional[EntityType] = None
    entity_id: Optional[UUID] = None
    action_type: NotificationAction = NotificationAction.NONE
    created_by: Optional[UUID] = None
    channels_sent: Optional[Dict[str, Any]] = None
    payload: Optional[Dict[str, Any]] = None

class NotificationResponse(NotificationBase):
    id: UUID
    recipient_id: UUID
    is_seen: bool
    is_read: bool
    is_deleted: bool
    created_at: datetime
    updated_at: Optional[datetime]

    model_config = ConfigDict(from_attributes=True)

class NotificationListResponse(BaseModel):
    items: List[NotificationResponse]
    total: int
    page: int
    size: int

class NotificationSummaryResponse(BaseModel):
    total: int
    unread: int
    unseen: int
    high_priority: int
    urgent: int

class NotificationSettingsBase(BaseModel):
    likes_enabled: bool = True
    messages_enabled: bool = True
    recruitment_enabled: bool = True
    crowdfunding_enabled: bool = True
    emails_enabled: bool = True

class NotificationSettingsResponse(NotificationSettingsBase):
    user_id: UUID
    updated_at: Optional[datetime]

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/notification.ts
export enum NotificationType {
    VIDEO_LIKED = "VIDEO_LIKED",
    VIDEO_COMMENTED = "VIDEO_COMMENTED",
    VIDEO_APPROVED = "VIDEO_APPROVED",
    VIDEO_REJECTED = "VIDEO_REJECTED",
    NEW_MESSAGE = "NEW_MESSAGE",
    MESSAGE_BLOCKED = "MESSAGE_BLOCKED",
    RECRUITMENT_REQUEST = "RECRUITMENT_REQUEST",
    RECRUITMENT_ACCEPTED = "RECRUITMENT_ACCEPTED",
    RECRUITMENT_REJECTED = "RECRUITMENT_REJECTED",
    RECRUITMENT_STAGE_CHANGED = "RECRUITMENT_STAGE_CHANGED",
    TALENT_SAVED = "TALENT_SAVED",
    DONATION_RECEIVED = "DONATION_RECEIVED",
    CAMPAIGN_COMPLETED = "CAMPAIGN_COMPLETED",
    CAMPAIGN_EXPIRED = "CAMPAIGN_EXPIRED",
    PROFILE_VERIFIED = "PROFILE_VERIFIED",
    ACCOUNT_SUSPENDED = "ACCOUNT_SUSPENDED",
    SECURITY_ALERT = "SECURITY_ALERT",
    PASSWORD_CHANGED = "PASSWORD_CHANGED",
    LOGIN_NEW_DEVICE = "LOGIN_NEW_DEVICE",
    SYSTEM = "SYSTEM"
}

export enum NotificationPriority {
    LOW = "LOW",
    NORMAL = "NORMAL",
    HIGH = "HIGH",
    URGENT = "URGENT"
}

export interface Notification {
    id: string;
    recipient_id: string;
    notification_type: NotificationType;
    priority: NotificationPriority;
    title: string;
    body: string;
    link?: string;
    is_seen: boolean;
    is_read: boolean;
    is_deleted: boolean;
    created_at: string;
    updated_at?: string;
}

export interface NotificationListResponse {
    items: Notification[];
    total: number;
    page: number;
    size: number;
}

export interface NotificationSettings {
    likes_enabled: boolean;
    messages_enabled: boolean;
    recruitment_enabled: boolean;
    crowdfunding_enabled: boolean;
    emails_enabled: boolean;
    user_id?: string;
    updated_at?: string;
}
